src/renderers/shared/reconciler/instantiateReactComponent.js

```diff
@@ -83,6 +106,8 @@ function instantiateReactComponent(node, parentCompositeType) {
       // represenations. I.e. ART. Once those are updated to use the string
       // representation, we can drop this code path.
       instance = new element.type(element);
+    } else if (isStatelessComponentType(element.type)) {
```

Comment:
@sebmarkbage I'm done with hack that supports stateless and module pattern components at same time.

Now component constructor called two times. I can fix this if you agreed to remove `context` argument from module pattern components constructor:
```js
// Now
type StatelessComponent = (props) -> ReactElement
type ModulePatternComponent = (props, context) -> ReactComponent

// Should be
type StatelessComponent = (props) -> ReactElement
type ModulePatternComponent = (props) -> ReactComponent
```